h2o-algos/src/main/java/hex/tree/DTree.java

```diff
@@ -1035,20 +1081,59 @@ static Split findBestSplitPoint(DHistogram hs, int col, double min_rows,
                   // Or tied SE, then pick split towards middle bins
                   (selo + sehi == best_seL + best_seR &&
                           Math.abs(b - (nbins >> 1)) < Math.abs(best - (nbins >> 1)))) {
-            if( (wlo[b] + wNA) >= min_rows && whi[b] >= min_rows) {
-              if (constraint == 0 || (constraint * (nomlo[b] + nomNA) / (denlo[b] + denNA) <= constraint * nomhi[b] / denhi[b])) {
+            if((wlo[b] + wNA) >= min_rows && whi[b] >= min_rows) {
+              double tmpPredLeft;
+              double tmpPredRight;
+              if(constraint != 0 && dist._family.equals(DistributionFamily.quantile)) {
+                  int quantileBinLeft = 0;
+                  int quantileBinRight = 0;
+                  double ratio = 1;
+                  double delta = 1;
+                  for (int bin = 1; bin <= nbins; bin++) {
+                    // left tree prediction quantile
+                    if (bin <= b) {
+                      if (bin == 1) {
+                        ratio = 100 / wlo[b];
```

Comment:
Where is the 100 number coming from?

Reply:
Good question! I calculated the percentile instead of quantile here. My mistake. I fixed it and run tests and demo and, it looks good. The calculation of the quantile is similar, so the performance should be the same.